lighthouse-core/report/html/renderer/element-screenshot-renderer.js

```diff
@@ -0,0 +1,240 @@
+/**
+ * @license Copyright 2020 The Lighthouse Authors. All Rights Reserved.
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License. You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions and limitations under the License.
+ */
+'use strict';
+
+/* globals self RectHelpers */
+
+/** @typedef {import('./dom.js')} DOM */
+/** @typedef {LH.Artifacts.Rect} Rect */
+/** @typedef {{width: number, height: number}} Size */
+
+/**
+ * @param {number} value
+ * @param {number} min
+ * @param {number} max
+ */
+function clamp(value, min, max) {
+  if (value < min) return min;
+  if (value > max) return max;
+  return value;
+}
+
+class ElementScreenshotRenderer {
+  /**
+   * @param {Rect} elementRectInScreenshotCoords
+   * @param {Size} elementPreviewSizeInScreenshotCoords
+   * @param {Size} screenshotSize
+   */
+  static getScreenshotPositions(
+      elementRectInScreenshotCoords, elementPreviewSizeInScreenshotCoords, screenshotSize) {
+    //
```

Comment:
```suggestion
```

Reply:
or add a comment to it? not sure what it's doing here :)

Reply:
> or add a comment to it? not sure what it's doing here :)

there's actually a few outstanding @patrickhulce comments that still seem worth addressing?

Reply:
https://github.com/GoogleChrome/lighthouse/pull/10716/files/c411fd8a5692d5ee86c763a07b7bf129fff3e0c2#r444532768